### PR TITLE
A few small fixes for serialize & register

### PR DESCRIPTION
**Pull request by katrogan**
# TL;DR

- Make it more clear that serialize terminated successfully
- Default to user's currently working dir when local source root is omitted in out of container serialize
- Raise an error when the expected tar file is missing during fast register

Tested locally.

**Note** I still can't recreate the

```
flytekit.common.exceptions.user.FlyteAssertion: No configuration set for [internal] image.  This is a required configuration.
```
issue or the case where a custom version is specified during fast register and tar file is not uploaded with the specified version as the filename.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Discovered while onboarding a new user.

## Tracking Issue
https://github.com/lyft/flyte/issues/<number>

## Follow-up issue
_NA_

### CODE REVIEW

File: flytekit/clis/sdk_in_container/serialize.py [hunk 1]
```diff
@@ -320,6 +321,15 @@ def fast_workflows(ctx, folder=None):
     if folder:
         click.echo(f"Writing output to {folder}")
 
+    source_dir = ctx.obj[CTX_LOCAL_SRC_ROOT]
```
Comment (katrogan) [inline flytekit/clis/sdk_in_container/serialize.py:324]: moved this up so the green success statement at the end of serialize prints last

**Review by wild-endeavor — APPROVED** [review]
